File: shared/src/design-tokens.ts
// AgentDeck — Design tokens (TS bindings)
// Mirror of design/tokens.css — see DESIGN.md for the spec.
// CSS file remains the source of truth; keep this file in sync when tokens change.

export const Tide = {
  s50: "#f5f3ec",
  s100: "#ebe6d6",
  s200: "#d8cfb6",
  s300: "#a8b09a",
} as const;

export const Ink = {
  s900: "#0e1f1f",
  s800: "#15302f",
  s700: "#1f4544",
  s500: "#426664",
  s300: "#7c9694",
} as const;

export const Kelp = {
  s700: "#1f6157",
  s500: "#2f8a7c",
  s300: "#6fb6a8",
} as const;

export const Coral = {
  s500: "#c0573a",
  s700: "#8c3a23",
} as const;

export const Amber = {
  s500: "#c8923a",
} as const;

// Marketing / editorial status semantics (DESIGN.md §2.7)
export const Status = {
  idle: Ink.s300,
  processing: Kelp.s500,
  awaiting: Amber.s500,
  error: Coral.s500,
} as const;

// Product UI palette — brighter signal colors for menubar / e-ink / hardware / TTY.
// DESIGN.md §2.6: marketing surfaces must NEVER use these.
export const UI = {
  ok: "#52D988",
  attn: "#FFA93D",
  error: "#FF6B6B",
  cyan: "#3ED6E8",
  idle: "#9a9aa2",
  idleDark: "#7a8a9c",
  popupBgDark: "#0a1a2a",
  popupBgDeep: "#061018",
  popupBgMid: "#0a1520",
  popupBgLight: "#f6f3ee",
  ttyBg: "#0c0d10",
  ttyBgMid: "#141820",
  ttyText: "#c8d0d8",
  ttyDim: "#7a8493",
  ttyFaint: "#4a5060",
} as const;

// Agent brand marks (only saturated reds/blues allowed in the system).
// Keep in sync with design/brand/*.svg.
export const Brand = {
  claudeCode: "#C07058",
  codex: "#6166E0",
  openclaw: "#FF4D4D",
  opencode: "#3a3a3a",
  antigravity: "#5F6368",
  kiro: "#7C3AED",
} as const;

export const Font = {
  sans: '"IBM Plex Sans", "IBM Plex Sans KR", "IBM Plex Sans JP", -apple-system, BlinkMacSystemFont, system-ui, sans-serif',
  mono: '"JetBrains Mono", "IBM Plex Mono", ui-monospace, monospace',
} as const;

export const Type = {
  hero: "clamp(54px, 7vw, 96px)",
  editorial: "clamp(56px, 8vw, 112px)",
  pageTitle: "clamp(38px, 5vw, 64px)",
  h2: "44px",
  h3: "26px",
  h3Lg: "32px",
  cardTitle: "19px",
  bodyLg: "19px",
  body: "17px",
  lede: "18px",
  small: "14.5px",
  caption: "13px",
  kicker: "12px",
  monoBadge: "11.5px",
} as const;

export const Tracking = {
  hero: "-0.035em",
  editorial: "-0.04em",
  h2: "-0.02em",
  h3: "-0.015em",
  card: "-0.01em",
  kicker: "0.18em",
  badge: "0.16em",
  chip: "0.08em",
} as const;

// Spacing scale (4px base) — DESIGN.md §4.1
export const Spacing = {
  s1: 4,
  s2: 8,
  s3: 12,
  s4: 16,
  s5: 20,
  s6: 24,
  s8: 32,
  s10: 40,
  s12: 48,
  s14: 56,
  s16: 64,
  s20: 80,
  s24: 96,
  s30: 120,
} as const;

export const Layout = {
  containerMax: 1240,
  containerPad: 32,
  sectionY: 96,
} as const;

export const Radius = {
  sm: 4,
  md: 8,
  lg: 10,
  xl: 12,
  xxl: 14,
  xxxl: 16,
  xxxxl: 18,
  pill: 999,
} as const;

export const Shadow = {
  card: "0 6px 20px -8px rgba(14, 31, 31, 0.45)",
  cardHover: "0 10px 28px -8px rgba(14, 31, 31, 0.55)",
  frame:
    "0 30px 80px -30px rgba(14, 31, 31, 0.4), 0 8px 30px -10px rgba(14, 31, 31, 0.18)",
  canvas: "0 30px 80px -20px rgba(0, 0, 0, 0.6)",
} as const;

export const Motion = {
  easeSnap: "cubic-bezier(0.2, 0.6, 0.2, 1)",
  fast: 120,
  base: 200,
  slow: 320,
  pulse: 1100,
  wiggle: 700,
} as const;

export const DesignTokens = {
  Tide,
  Ink,
  Kelp,
  Coral,
  Amber,
  Status,
  UI,
  Brand,
  Font,
  Type,
  Tracking,
  Spacing,
  Layout,
  Radius,
  Shadow,
  Motion,
} as const;

File: apple/AgentDeck/UI/Common/DesignTokens.swift
// DesignTokens.swift — Swift mirror of design/tokens.css.
// See DESIGN.md for the spec. The CSS file remains the source of truth;
// keep this file in sync when tokens change.

import SwiftUI

private func tokenColor(_ hex: String) -> Color {
    var s = hex
    if s.hasPrefix("#") { s.removeFirst() }
    guard s.count == 6, let v = UInt32(s, radix: 16) else {
        return Color(red: 1.0, green: 0.0, blue: 1.0)
    }
    let r = Double((v >> 16) & 0xff) / 255.0
    let g = Double((v >> 8) & 0xff) / 255.0
    let b = Double(v & 0xff) / 255.0
    return Color(red: r, green: g, blue: b)
}

enum DesignTokens {

    // MARK: - Tide (sand / paper)

    enum Tide {
        static let s50  = tokenColor("#f5f3ec")
        static let s100 = tokenColor("#ebe6d6")
        static let s200 = tokenColor("#d8cfb6")
        static let s300 = tokenColor("#a8b09a")
    }

    // MARK: - Ink (deep aquarium)

    enum Ink {
        static let s900 = tokenColor("#0e1f1f")
        static let s800 = tokenColor("#15302f")
        static let s700 = tokenColor("#1f4544")
        static let s500 = tokenColor("#426664")
        static let s300 = tokenColor("#7c9694")
    }

    // MARK: - Kelp (App Store / running / OK)

    enum Kelp {
        static let s700 = tokenColor("#1f6157")
        static let s500 = tokenColor("#2f8a7c")
        static let s300 = tokenColor("#6fb6a8")
    }

    // MARK: - Coral (Developer / build)

    enum Coral {
        static let s500 = tokenColor("#c0573a")
        static let s700 = tokenColor("#8c3a23")
    }

    // MARK: - Amber (attention only — only color allowed to pulse)

    enum Amber {
        static let s500 = tokenColor("#c8923a")
    }

    // MARK: - Marketing status semantics (DESIGN.md §2.7)

    enum Status {
        static let idle       = Ink.s300
        static let processing = Kelp.s500
        static let awaiting   = Amber.s500
        static let error      = Coral.s500
    }

    // MARK: - Product UI palette (menubar / e-ink / hardware / TTY)
    // Brighter signal colors. DESIGN.md §2.6: marketing surfaces must NEVER use these.

    enum UI {
        static let ok           = tokenColor("#52D988")
        static let attn         = tokenColor("#FFA93D")
        static let error        = tokenColor("#FF6B6B")
        static let cyan         = tokenColor("#3ED6E8")
        static let idle         = tokenColor("#9a9aa2")
        static let idleDark     = tokenColor("#7a8a9c")
        static let popupBgDark  = tokenColor("#0a1a2a")
        static let popupBgDeep  = tokenColor("#061018")
        static let popupBgMid   = tokenColor("#0a1520")
        static let popupBgLight = tokenColor("#f6f3ee")
        static let ttyBg        = tokenColor("#0c0d10")
        static let ttyBgMid     = tokenColor("#141820")
        static let ttyText      = tokenColor("#c8d0d8")
        static let ttyDim       = tokenColor("#7a8493")
        static let ttyFaint     = tokenColor("#4a5060")
    }

    // MARK: - Agent brand marks
    // The only saturated reds/blues allowed in the system. Sourced from upstream
    // brand SVGs in design/brand/ — do not redraw or restyle.

    enum Brand {
        static let claudeCode = tokenColor("#C07058")
        static let codex      = tokenColor("#6166E0")
        static let openclaw   = tokenColor("#FF4D4D")
        static let opencode   = tokenColor("#3a3a3a")
        static let antigravity = tokenColor("#5F6368")
        static let kiro       = tokenColor("#7C3AED")
    }

    // MARK: - Type stack

    enum Font {
        static let sans = "IBM Plex Sans"
        static let sansKR = "IBM Plex Sans KR"
        static let sansJP = "IBM Plex Sans JP"
        static let mono = "JetBrains Mono"
        static let monoFallback = "IBM Plex Mono"
    }

    // MARK: - Type scale (DESIGN.md §3.2)

    enum FontSize {
        static let h2: CGFloat         = 44
        static let h3: CGFloat         = 26
        static let h3Lg: CGFloat       = 32
        static let cardTitle: CGFloat  = 19
        static let bodyLg: CGFloat     = 19
        static let body: CGFloat       = 17
        static let lede: CGFloat       = 18
        static let small: CGFloat      = 14.5
        static let caption: CGFloat    = 13
        static let kicker: CGFloat     = 12
        static let monoBadge: CGFloat  = 11.5
    }

    enum Tracking {
        static let hero: CGFloat       = -0.035
        static let editorial: CGFloat  = -0.04
        static let h2: CGFloat         = -0.02
        static let h3: CGFloat         = -0.015
        static let card: CGFloat       = -0.01
        static let kicker: CGFloat     = 0.18
        static let badge: CGFloat      = 0.16
        static let chip: CGFloat       = 0.08
    }

    // MARK: - Spacing scale (4px base)

    enum Spacing {
        static let s1: CGFloat  = 4
        static let s2: CGFloat  = 8
        static let s3: CGFloat  = 12
        static let s4: CGFloat  = 16
        static let s5: CGFloat  = 20
        static let s6: CGFloat  = 24
        static let s8: CGFloat  = 32
        static let s10: CGFloat = 40
        static let s12: CGFloat = 48
        static let s14: CGFloat = 56
        static let s16: CGFloat = 64
        static let s20: CGFloat = 80
        static let s24: CGFloat = 96
        static let s30: CGFloat = 120
    }

    // MARK: - Radii

    enum Radius {
        static let sm: CGFloat   = 4
        static let md: CGFloat   = 8
        static let lg: CGFloat   = 10
        static let xl: CGFloat   = 12
        static let xxl: CGFloat  = 14
        static let xxxl: CGFloat = 16
        static let xxxxl: CGFloat = 18
        static let pill: CGFloat = 999
    }

    // MARK: - Layout

    enum Layout {
        static let containerMax: CGFloat = 1240
        static let containerPad: CGFloat = 32
        static let sectionY: CGFloat     = 96
    }

    // MARK: - Motion (durations in milliseconds — divide by 1000 for seconds)

    enum Motion {
        static let fast: Int   = 120
        static let base: Int   = 200
        static let slow: Int   = 320
        static let pulse: Int  = 1100
        static let wiggle: Int = 700
    }
}
